components/org.wso2.micro.integrator.initializer/src/main/java/org/wso2/micro/integrator/initializer/deployment/application/deployer/CappDeployer.java

```diff
@@ -193,6 +213,12 @@ private void deployCarbonApps(String artifactPath) throws CarbonException {
             faultyCAppObjects.add(currentApp);
             faultyCapps.add(cAppName);
         }
+        if (ServiceCatalogUtils.isServiceCatalogEnabled()) {
```

Comment:
We shouldn't run this if Deployment exception occurs in the above step right ?
Else we will run this for faulty CAPPs also.

Reply:
When we execute the ServiceCatalogDeployer, it will check whether some preconditions are satisfied. One of the preconditions is to identify whether there are any faultyCapps. If there are faultyCapps, it will print the relevant log message and abort uploading to Service Catalog.

Reply:
Added a check to avoid spawning a thread every time for FaultyCApps.